x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -25,10 +26,30 @@ import {
   InferencePipelineInferenceConfig,
 } from '../types/pipelines';
 
-export const TEXT_EXPANSION_TYPE = SUPPORTED_PYTORCH_TASKS.TEXT_EXPANSION;
-export const TEXT_EXPANSION_FRIENDLY_TYPE = 'ELSER';
-export const ML_INFERENCE_PREFIX = 'ml.inference.';
-export const ELSER_MODEL_ID = '.elser_model_1';
+const ML_INFERENCE_PREFIX = 'ml.inference.';
+
+class InferenceModel {
+  public taskType: string;
+  public friendlyTaskType: string;
+  public modelId: string;
+
+  constructor() {
+    this.taskType = SUPPORTED_PYTORCH_TASKS.TEXT_EXPANSION;
+    this.friendlyTaskType = 'ELSER';
+    this.modelId = '';
+  }
+
+  public setModelId(modelId: string): void {
+    this.modelId = modelId;
+  }
+}
+
+export const elserInferenceModel = new InferenceModel();
```

Comment:
I'm not sure we'd want to store state locally like this, especially for the frontend. If we're updating this to not use constants then we'll likely need to put this in a store for the frontend.

I don't think we do this elsewhere already, cc: @sphilipse